src/templates/cmp.js

```diff
@@ -0,0 +1,175 @@
+    var logCallbackIndex = -1;
+    var iframe;
+
+    function message(message, callback) {
+      if (!iframe.contentWindow) {
+        return;
+      }
+      var params = message.split(';');
+      callbackMap[++callbackCount] = [callback];
+
+      if (params[1] === 'onLogEvent') {
+        logCallbackIndex = callbackCount;
+        logQueue();
+      }
+      iframe.contentWindow.postMessage(callbackCount + ';' + message, '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>');
+    }
+
+    function log(event, success, parameters) {
+      window.__tcfapi('log', 2, function () {}, btoa(JSON.stringify({ event: event, parameters: parameters })));
+    }
+
+    function createIframe() {
+      var iframe = document.createElement('iframe');
+
+      iframe.setAttribute(
+        'src',
+        '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/iframe.html' +
+          (channelId !== '' ? '?channelId=' + channelId : ''),
+      );
+      iframe.setAttribute('style', 'border:0;outline:0;width:0;height:0;');
+      iframe.setAttribute('frameborder', '0');
+
+      iframe.addEventListener('load', function () {
+        window.__tcfapi = function (command, version, callback, parameter) {
+          message('cmd;' + command + ';' + version + ';' + parameter, callback);
+        };
+
+        window.addEventListener(
+          'message',
+          function (ev) {
+            try {
+              if (ev.origin === '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>' && ev.data) {
+                var message = ev.data.split(';');
+                var position = message[0] === 'err' ? 1 : 0;
+                var id = message[position];
+                var callback = callbackMap[id][position];
+                if (logCallbackIndex + '' !== id) delete callbackMap[id];
+                var r = JSON.parse(atob(message[++position]));
+                var s = message[++position] ? true : undefined;
+                if (callback) {
+                  callback(r, s);
+                }
+              }
+            } catch (e) {}
+          },
+          false,
+        );
+
+        onAPILoaded('iframe');
+      });
+
+      iframe.addEventListener('error', log.bind(null, 'loaded', false, { type: 'iframe' }));
+
+      return iframe;
+    }
+
+    var iframeRetries = 0;
+
+    function loadIframe() {
+      if (iframeRetries >= 3) {
+        return;
+      }
+
+      if (!document.body) {
+        setTimeout(function () {
+          iframeRetries++;
+          loadIframe();
+        }, 100);
+        return;
+      }
+
+      iframe = createIframe();
+      document.body.appendChild(iframe);
+    }
+
+    function createTcfapiScriptTag() {
+      var xt = '<%-XT%>';
+      var hasConsent;
+
+      if (window.localStorage && localStorage.getItem && localStorage.setItem) {
+        if (!localStorage.getItem('<%-TECH_COOKIE_NAME%>')) {
+          // no tech info yet, need to store
+          localStorage.setItem('<%-TECH_COOKIE_NAME%>', xt);
```

Comment:
I think this will set an entry on the channel's hbbTv app domain, which we don't want.